modules/k3s/src/test/java/org/testcontainers/k3s/KubectlContainerTest.java

```diff
@@ -0,0 +1,54 @@
+package org.testcontainers.k3s;
+
+import org.assertj.core.api.Assertions;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.startupcheck.OneShotStartupCheckStrategy;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+
+public class KubectlContainerTest {
+
+    public static Network network = Network.SHARED;
+
+    @ClassRule
+    public static K3sContainer k3s = new K3sContainer(DockerImageName.parse("rancher/k3s:v1.21.3-k3s1"))
+        .withNetwork(network)
+        .withNetworkAliases("k3s");
+
+    @Test
+    public void shouldExposeKubeConfigForNetworkAlias() throws Exception {
+
+        String kubeConfigYaml = k3s.getKubeConfigYaml("k3s");
+
+        Path tempFile = Files.createTempFile(null, null);
+        Files.write(tempFile, kubeConfigYaml.getBytes(StandardCharsets.UTF_8));
```

Comment:
Tangential thought for @bsideup, @kiview: it seems an interesting missed opportunity that `withCopyFileToContainer` doesn't have a way to accept a `String/byte[]/Supplier` for the file contents 😂 
Maybe we should add that at some point...

Reply:
Maybe 😂 
https://github.com/testcontainers/testcontainers-java/issues/3814

Reply:
(funny how yesterday was 1 year anniversary of that ticket 😅)

Reply:
And there is a corresponding PR 😅 
https://github.com/testcontainers/testcontainers-java/pull/3815

Reply:
LOL brilliant 😆

Reply:
Let's look at #3815 again some time but not let it block this PR.

Reply:
The idea needed 1 year to mature :)

Reply:
FTR the PR stopped at japicmp valid complains about the backward compatibility being broken, so we need to restore it.